backend/typescript/graphql/index.ts

```diff
@@ -25,4 +31,29 @@ const executableSchema = makeExecutableSchema({
   resolvers: merge(authResolvers, entityResolvers, userResolvers),
 });
 
-export default executableSchema;
+const authorizedByAllRoles = () =>
+  isAuthorizedByRole(new Set(["User", "Admin"]));
+const authorizedByAdmin = () => isAuthorizedByRole(new Set(["Admin"]));
+
+const graphQLMiddlewares = {
+  Query: {
+    entity: authorizedByAllRoles(),
+    entities: authorizedByAllRoles(),
+    userById: authorizedByAdmin(),
+    userByEmail: authorizedByAdmin(),
+    users: authorizedByAdmin(),
+  },
+  Mutation: {
+    createEntity: authorizedByAllRoles(),
+    updateEntity: authorizedByAllRoles(),
+    deleteEntity: authorizedByAllRoles(),
+    createUser: authorizedByAdmin(),
+    updateUser: authorizedByAdmin(),
+    deleteUserById: authorizedByAdmin(),
+    deleteUserByEmail: authorizedByAdmin(),
+    logout: isAuthorizedByUserId("userId"),
+    resetPassword: isAuthorizedByEmail("email"),
```

Comment:
Yeah doing it this way instead of in the business logic actually seems to make more sense to me for what we have right now